client/lib/cart-values/cart-items.js

```diff
@@ -1308,7 +1308,27 @@ export function isPaidDomain( domainPriceRule ) {
 	return 'PRICE' === domainPriceRule;
 }
 
-export function getDomainPriceRule( withPlansOnly, selectedSite, cart, suggestion, isDomainOnly ) {
+const isMonthlyOrFreeFlow = ( flowName ) => {
+	return (
+		flowName &&
+		[
+			'free',
+			'personal-monthly',
+			'premium-monthly',
+			'business-monthly',
+			'ecommerce-monthly',
+		].includes( flowName )
+	);
+};
```

Comment:
While the code in general looks good to me, using a flow name here somewhat makes me cautious because I sort of recall we have seen some issues causing by refreshing and back navigation in the past. However, I'm not sure if that's still the case. @niranjan-uma-shankar could you help me confirm if this is safe?

Reply:
@southp I don't exactly recollect it. In this case, since we're restricting by flow name, I believe it to be safe.